peerconnection.go

```diff
@@ -1014,6 +1014,12 @@ func (pc *PeerConnection) SetRemoteDescription(desc SessionDescription) error {
 	detectedPlanB := descriptionIsPlanB(pc.RemoteDescription())
 	weOffer := desc.Type == SDPTypeAnswer
 
+	for _, tr := range localTransceivers {
```

Comment:
Do we need this? I thought `SetCodecPreferences` was for a user to explicitly set codecs only.

Reply:
@digitalix

Reply:
@Sean-Der I think we do. This was needed in case if a peer uses different payload ids for codecs. For example, if the payload id for VP8 in Safari is 100 but in Chrome it's 96. So it will fail if a user uses SetCodecPreferences before SetRemoteDescription.

Reply:
Alternatively, we can set new codecs as "pending" and apply them during re-negotiation. I think this is how it works in libwebrtc.

Reply:
@digitalix ah I seel

So I think the right fix is to update `getCodecs` to pull the proper PayloadTypes from the `MediaEngine`?

Similiar to what `SetCodecPreferences` does right now. Maybe we can even drop the input sanitization in `SetCodecPreferences` and filter invalid codecs in getCodecs?

If that is right I can implement, add a test and merge right now!

Reply:
@Sean-Der I think I should have placed a comment there to clarify why it's there :). 

```So I think the right fix is to update getCodecs to pull the proper PayloadTypes from the MediaEngine?```
This is how I had it at the beginning, but I didn't like it that it would happen every time we call `getCodecs`. So I came up with something more elegant IMO. If that's fine then go ahead! :) 

```Maybe we can even drop the input sanitization in SetCodecPreferences```
We can, but this would be different to SetCodecPreferences in specs. 
https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpTransceiver/setCodecPreferences

Additionally, now the user wouldn't know if it actually worked until it's negotiated, but `SetCodecPreferences` in browsers would fail immediately.

Reply:
Oh great point about the error handling!

Ok I am going to make the change to `getCodecs` and write a test that confirms filtering + PayloadTypes are correct.

Really fantastic work on this PR, this is a fantastic change.

Reply:
Thanks!